web_api.md

```diff
@@ -0,0 +1,261 @@
+   The `ssh` section behaves exactly the same as described in the `message_queue` section above.
+3. `hosts`: this section contains a list of hosts, usually virtual machines, where Processing Servers should be
+   deployed. To be able to connect to a host, an `address` and `username` are required, then comes either `password`
+   or `path_to_privkey` (path to a private key). All Processing Servers, which will be deployed, must be declared under
+   the `deploy_processors` property. In case `type` is `docker`, make sure that [Docker](https://www.docker.com/) is
+   installed in the target machine and the provided `username` has enough rights to execute Docker commands.
+
+Among three sections, only the `message_queue` is required. However, if `hosts` is presented, `mongo_db` must be there
+as well. For more information, please check the [configuration file schema](web_api/config.schema.yml).
+
+### Processing Server
+
+One do not need to start (or stop) a Processing Server manually, since it can be managed by a Processing Broker via
+a [configuration file](#processing-broker). However, if it is necessary to do so, there are two ways to start a
+Processing Server:
+
+```shell
+# 1. Use ocrd tool
+$ ocrd processing-server <processor-name> --queue=<queue-address> --database=<database-address>
+
+# 2. Use processor name
+<processor-name> processing-server --queue=<queue-address> --database=<database-address>
+```
+
+* `--queue`: a [Rabbit MQ connection string](https://www.rabbitmq.com/uri-spec.html) to a running instance.
+* `--database`: a [MongoDB connection string](https://www.mongodb.com/docs/manual/reference/connection-string/) to a
+  running instance.
+
+### Message Queue
+
+By using a queuing system, specifically [RabbitMQ](https://www.rabbitmq.com/), the reliability and flexibility of the
+whole system are greatly improved. In our implementation, manual acknowledgement mode is used. This means, when a
+Processing Server finishes successfully, it sends a positive ACK signal to RabbitMQ. In case of failure, it tries again
+three times before sending a negative ACK signal. When a negative signal is received, RabbitMQ will re-queue the
+message. If there is not any ACK signal sent for any reason (e.g. consumer crash, power outage, network problem, etc.),
+RabbitMQ will automatically re-queue the message after timeout, which is 30
+minutes [by default](https://www.rabbitmq.com/consumers.html#acknowledgement-timeout). This value is, however, can be
+configured.
+
+To avoid processing the same input twice (in case of re-queuing), a Processing Server first checks
+the [`redeliver`](https://www.rabbitmq.com/confirms.html#automatic-requeueing) property to see if this message was
+re-queued. If yes, and the status of this process in the database is not `SUCCESS`, it will process the data described
+in the message again.
+
+When a Processing Broker receives a request, it adds some more data to the content, then push it to an appropriate
```

Comment:
>it adds some more data to the content

What is "some more data" here?

Reply:
Or rather, as it is explained later, maybe remove this here to reduce confusion.

Reply:
I re-phrased it.